api/src/main/java/org/openmrs/module/metrics/api/db/hibernate/HibernateMetricsDAO.java

```diff
@@ -15,69 +15,66 @@
-	
+
 	/**
 	 * @param sessionFactory the sessionFactory to set
 	 */
 	@Autowired
 	public void setSessionFactory(DbSessionFactory sessionFactory) {
 		this.sessionFactory = sessionFactory;
 	}
-	
+
 	@Override
 	public MetricEvent saveMetricEvent(MetricEvent metricEvent) {
 		sessionFactory.getCurrentSession().saveOrUpdate(metricEvent);
 		return metricEvent;
 	}
-	
+
 	@Override
-	public Integer getEncounterObjectsByGivenDateRangeAndType(LocalDateTime startRange, LocalDateTime endRange,
-	        String encounterType) {
+	public Integer getEncounterObjectsByGivenDateRangeAndType(LocalDateTime startRange, LocalDateTime endRange, String encounterType) {
 		// @formatter:off
 		String sql = "select count(*) "
-		        + "from event_records e "
-		        + "inner join encounter b on e.object_uuid = b.uuid "
-		        + "inner join encounter_type c on b.encounter_type = c.encounter_type_id "
-		        + "where e.title = 'Encounter' and (e.date_created between :startRange and :endRange) and e.tags = 'CREATED' "
-		        + "and c.name = :encounterType";
+				+ "from metric_event_records e "
+				+ "inner join encounter b on e.object_uuid = b.uuid "
+				+ "inner join encounter_type c on b.encounter_type = c.encounter_type_id "
+				+ "where e.title = 'Encounter' and (e.date_created between :startRange and :endRange) and e.tags = 'CREATED' "
+				+ "and c.name = :encounterType";
 		// @formatter:on
-		
-		return (Integer) sessionFactory.getCurrentSession().createSQLQuery(sql)
-		        .setParameter("startRange", Date.from(startRange.atZone(ZoneId.systemDefault()).toInstant()))
-		        .setParameter("endRange", Date.from(endRange.atZone(ZoneId.systemDefault()).toInstant()))
-		        .setParameter("encounterType", encounterType).uniqueResult();
+
+		return (Integer) sessionFactory.getCurrentSession().createSQLQuery(sql).setParameter("startRange", startRange)
+				.setParameter("endRange", endRange).setParameter("encounterType", encounterType).uniqueResult();
 	}
-	
+
 	@Override
 	public Integer getNewPatientsObjectsByGivenDateRange(LocalDateTime startRange, LocalDateTime endRange) {
-		// @formatter:off
-		String sql = "select count(*) " + "from event_records e " + "where e.title='Patient' and "
-		        + "(e.date_created between :startRange and :endRange) and " + "e.tags = 'CREATED'";
+		// @formatter:Date
+		String sql = "select count(*) " + "from metric_event_records e " + "where e.title='Patient' and "
+				+ "(e.date_created between :startRange and :endRange) and " + "e.tags = 'CREATED'";
 		// @formatter:on
-		
-		return (Integer) sessionFactory.getCurrentSession().createSQLQuery(sql)
-		        .setParameter("startRange", Date.from(startRange.atZone(ZoneId.systemDefault()).toInstant()))
-		        .setParameter("endRange", Date.from(endRange.atZone(ZoneId.systemDefault()).toInstant())).uniqueResult();
+
+		return ((BigInteger) sessionFactory.getCurrentSession().createSQLQuery(sql).setParameter("startRange", startRange)
```

Comment:
Why `BigInteger` here?

Reply:
so result is bigint and finally am converting it to the integer value 
```
	return ((BigInteger) sessionFactory.getCurrentSession().createSQLQuery(sql).setParameter("startRange", startRange)
				.setParameter("endRange", endRange).uniqueResult()).intValue();
```